lib/src/blocs/comment/comment_bloc.dart

```diff
@@ -78,22 +78,56 @@ class CommentBloc extends Bloc<CommentEvent, CommentState> {
 
     // Sort comment by date
     on<SortCommentByDate>((event, emit) {
-      listCommentModel.comments.sort((comment1, comment2) =>
-          convertDate(comment1.Date).compareTo(convertDate(comment2.Date)));
-      listCommentModel.comments = listCommentModel.comments.reversed.toList();
+      listCommentModel.comments.sort((comment1, comment2) {
+        if (comment2.isPin == comment1.isPin) {
+          return convertDate(comment2.Date)
+              .compareTo(convertDate(comment1.Date));
+        } else {
+          return comment2.isPin! ? 1 : -1; // true values come first
+        }
+      });
+
```

Comment:
why do you need to return 1 or -1, I think in actual return a boolean is the best way

Reply:
This function does not set "comment.isPin" to True or False but indicates the order of the two comments being compared (sort by comparing which one should come before). And importantly, they required integer values, so if use a boolean instead, the function would no longer work correctly.

Reply:
ok, It sounds reasonable!